dashboard/app/models/script.rb

```diff
@@ -1293,11 +1293,40 @@ def self.with_default_fields
     Script.includes(:levels, :script_levels, lessons: :script_levels)
   end
 
+  def get_lesson_groups_i18n(lesson_groups_data)
+    lessons_data = lesson_groups_data.map {|lg| lg['lessons']}.flatten
+
+    # Do not write the names of existing lessons. Once a lesson has been
+    # created, its name is owned by the lesson edit page.
+    lessons_i18n = lessons_data.reject {|l| l['id']}.map do |lesson_data|
+      [lesson_data['key'], {name: lesson_data['name']}]
+    end.to_h
+
+    lesson_groups_i18n = lesson_groups_data.select {|lg| lg['user_facing']}.map do |lg_data|
```

Comment:
I think if a lesson group is deleted, it's strings will stick around. Is this okay?

Reply:
I'm not sure how desirable this behavior is, but I believe this is the same as the existing behavior both for lessons and lesson groups. There may be an argument in favor of keeping the string since we may have already paid to translate it, but I'm not certain of that. I'd be happy to defer to you, foundations, or anyone else familiar with translations on what we should do here.

Reply:
I'm not sure if it's desirable per se but I doubt it's a big deal, especially if that's the existing behavior.